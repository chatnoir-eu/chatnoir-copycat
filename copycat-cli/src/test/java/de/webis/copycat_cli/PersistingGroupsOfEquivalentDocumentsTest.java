package de.webis.copycat_cli;

import java.io.File;
import java.nio.file.Files;
import java.nio.file.Path;

import org.approvaltests.Approvals;
import org.junit.Test;

import lombok.SneakyThrows;
import net.sourceforge.argparse4j.inf.Namespace;

public class PersistingGroupsOfEquivalentDocumentsTest {

	@Test
	public void approveSmallSampleWithMultipleTopicsAndSmallGroups() {
		Approvals.verify(runOnFile("small-sample-multiple-topics-only-small-groups.jsonl"));
	}
	
	@Test
	public void approveSmallSampleWithMultipleTopicsAndLargeGroups() {
		Approvals.verify(runOnFile("multiple-topics-larger-groups.jsonl"));
	}
	
	@Test
	public void approveToucheExampleTopicTask1() {
		Approvals.verify(runOnFile("touche21-topic91.jsonl"));
	}
	
	@SneakyThrows
	String runOnFile(String inputFile) {
		Path outputFile = Files.createTempFile("unit-test-copycat", ".jsonl");
		Namespace args = argsForInputFile("src/test/resources/test-pairwise-near-duplicates/" +inputFile, outputFile.toFile());
		App.persistGroupsOfEquivalentDocuments(args);
		String ret = new String(Files.readAllBytes(outputFile));
		outputFile.toFile().delete();
		
		return ret;
	}
	
	private static Namespace argsForInputFile(String inputFile, File outputFile) {
		return CliArguments.parseArgs(new String[] {
			"--documents", "AnseriniIndex",
			"--output", inputFile,
			"--input", "a",
			"--documentEquivalenceClassFile", outputFile.toString()
		});
	}
}
