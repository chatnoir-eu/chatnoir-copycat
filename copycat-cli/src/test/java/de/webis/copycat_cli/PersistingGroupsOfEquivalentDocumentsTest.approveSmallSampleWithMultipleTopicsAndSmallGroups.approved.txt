
{"hash":"-639055371","ids":["doc-1","doc-2"]}
{"hash":"41003061","ids":["doc-3","doc-4"]}
{"hash":"721061493","ids":["doc-5","doc-6"]}
{"hash":"-639055371","ids":["doc-1","doc-2"]}
{"hash":"-1766451371","ids":["doc-4","doc-5"]}